OneCryptoPkg/OneCryptoLoaders/OneCryptoLoaderDxe.c

```diff
@@ -0,0 +1,489 @@
+    return Status;
+  }
+
+  //
+  // Confirm that the image is an EFI driver (really should be a MM_STANDLONE_DRIVER)
+  //
+  if (Image.Context.ImageType != EFI_IMAGE_SUBSYSTEM_EFI_BOOT_SERVICE_DRIVER) {
+    DEBUG ((DEBUG_ERROR, "Invalid image type: %d\n", Image.Context.ImageType));
+    return EFI_UNSUPPORTED;
+  }
+
+  //
+  // Grab the export directory from the loaded image
+  //
+  Status = GetExportDirectoryInPeCoffImage (&Image, &Exports);
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "Failed to get export directory from loaded image: %r\n", Status));
+    return Status;
+  }
+
+  DEBUG_CODE_BEGIN ();
+
+  //
+  // Print out the exported functions for debugging
+  //
+  PrintExportedFunctions (&Image, Exports);
+
+  DEBUG_CODE_END ();
+
+  //
+  // Find the constructor function
+  //
+  Status = FindExportedFunction (&Image, Exports, EXPORTED_CONSTRUCTOR_NAME, &RVA);
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "Failed to find exported function '%a': %r\n", EXPORTED_CONSTRUCTOR_NAME, Status));
+    return Status;
+  }
+
+  //
+  // Setup the Library constructor function
+  // Since the image is already loaded and relocated, we can directly use the RVA
+  //
+  *Constructor = (CONSTRUCTOR)((EFI_PHYSICAL_ADDRESS)LoadedImage->ImageBase + RVA);
+
+  DEBUG ((
+    DEBUG_ERROR,
+    "Crypto Constructor found at address: %p (Base: %p + RVA: 0x%x)\n",
+    *Constructor,
+    LoadedImage->ImageBase,
+    RVA
+    ));
+
+  return EFI_SUCCESS;
+}
+
+/**
+ * Entry point for the DXE (Driver Execution Environment) phase.
+ *
+ * This function is the main entry point for the DXE phase of the UEFI (Unified Extensible Firmware Interface) firmware.
+ * It is responsible for initializing the DXE environment and executing the DXE drivers.
+ *
+ * @param ImageHandle  The firmware allocated handle for the EFI image.
+ * @param SystemTable  A pointer to the EFI System Table.
+ *
+ * @retval EFI_SUCCESS           The entry point is executed successfully.
+ * @retval EFI_LOAD_ERROR        Failed to load the DXE environment.
+ * @retval EFI_INVALID_PARAMETER One or more parameters are invalid.
+ */
+EFI_STATUS
+EFIAPI
+DxeEntryPoint (
+  IN EFI_HANDLE        ImageHandle,
+  IN EFI_SYSTEM_TABLE  *SystemTable
+  )
+{
+  //
+  // Add debug output as the very first thing to see if we reach the entry point
+  //
+  if (SystemTable != NULL && SystemTable->ConOut != NULL) {
+    SystemTable->ConOut->OutputString(SystemTable->ConOut, L"*** DXE ENTRY POINT REACHED ***\r\n");
+  }
+
+  EFI_STATUS                 Status;
+  VOID                       *SectionData;
+  UINTN                      SectionSize;
+  CONSTRUCTOR                Constructor;
+  EFI_LOADED_IMAGE_PROTOCOL  *LoadedImage;
+  EFI_HANDLE                 LoadedImageHandle;
+
+  LoadedImageHandle = NULL;
+  LoadedImage       = NULL;
+
+  DEBUG ((DEBUG_ERROR, "OneCryptoLoaderDxe: Entry point called\n"));
+
+  //
+  // This must match the INF for OneCryptoBin
```

Comment:
This seems like it should be defined in the dec and used in both places from there (not sure if you can use in an INF)